### PR TITLE
store/groupcache - use updated twitter/groupcache library that supports proper cache population

### PR DESCRIPTION
**Pull request by dgdonut**
Refactor groupcache Write as a groupcache "PutterFunc", see https://github.com/twitter-forks/groupcache/pull/1 for context.

groupcache submodule branch will be updated to a real commit when above PR lands.

Keep travis go version on latest minor release.

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -65,7 +65,7 @@
 [submodule "github.com/twitter/groupcache"]
 	path = vendor/github.com/twitter/groupcache
 	url = https://github.com/twitter/groupcache
-	branch = 85fde1c5c36988481c30fc26605cd83f21cc94d1
+	branch = dgassaway/groupcache-put
```
Comment (jasonbs10) [inline .gitmodules]: Reminder to change.

File: snapshot/store/groupcache_store.go [hunk 2]
```diff
@@ -57,8 +63,26 @@ func MakeGroupcacheStore(underlying Store, cfg *GroupcacheConfig, stat stats.Sta
...
-		},
-	))
+		}),
+		groupcache.PutterFunc(func(ctx groupcache.Context, bundleName string, data []byte) error {
+			log.Info("New bundle, write and populate cache: ", bundleName)
+			stat.Counter(stats.GroupcacheWriteUnderlyingCounter).Inc(1)
+			defer stat.Latency(stats.GroupcacheWriteUnderlyingLatency_ms).Time().Stop()
+
+			// Get TTL via context type assertion, or set to nil and let underlying use its default
+			ttlc, ok := ctx.(*ttlContext)
```
Comment (jasonbs10) [inline snapshot/store/groupcache_store.go:73]: Tests to make sure this is propagated correctly?

Comment (dgdonut) [reply snapshot/store/groupcache_store.go:73]: We could test this in scoot, but it doesn't really belong here. The functionality is tested in the groupcache repo. A test in scoot could also pass as a false positive if e.g. the consistent hash changed.

**Comment by dgdonut** [comment]
merging per offline approval from @jasonbs10

**Pull request merged**